internal/service/kubernetes/node_group.go

```diff
@@ -389,7 +389,7 @@ func (r *kubernetesNodeGroupResource) Schema(_ context.Context, _ resource.Schem
 							MarkdownDescription: "Taint key.",
 							Required:            true,
 							Validators: []validator.String{
-								stringvalidator.RegexMatches(regexp.MustCompile("^[a-zA-Z0-9-]+$"), "needs to match regexp ^[a-zA-Z0-9-]+$"),
+								stringvalidator.RegexMatches(utils.ValidLabelKeyRegExp, utils.InvalidLabelKeyMessage),
```

Comment:
Instead of using these from the `utils` package, it'd be a bit simpler to introduce the variable and the constant in this file, for example:

```
var taintKeyRegexpStr = regexp.MustCompile("^[ -^`-~]+[ -~]*$") // Printable ASCII characters: ' ' (Space), ..., '^', '_', '`', ..., `~`
```

Reply:
Sure. My reasoning was to ensure the validation regex would be the same for taint keys and labels as apparently they follow the same syntax (though I couldn't find any definitive source for that). But keeping them separate also makes sense in case they aren't actually identical and require tweaking later on.

Will separate the two.